src/init.rs

```diff
@@ -360,3 +396,52 @@ pub fn init() -> ! {
         }
     }
 }
+
+// mod_logger needs to be called separately per module
+fn setup_stage2_init_fallback_log(fallback_log_filename: &str) {
+    info!(
+        "stage2-init:: Setting up temporary log destination to /tmp/{}",
+        fallback_log_filename
+    );
+    Logger::set_color(false);
+
+    let log_file = open_fallback_log_file(fallback_log_filename);
+
+    if log_file.is_some() {
+        match Logger::set_log_dest(&LogDestination::StreamStderr, log_file) {
+            Ok(_) => (),
+            Err(_) => {
+                error!(
+                    "Could not set logging to tmpfs at /tmp/{}",
+                    fallback_log_filename
+                )
+            }
+        }
+    }
+    Logger::flush();
+    sync();
+    info!(
+        "stage2-init:: Now logging to /tmp/{}",
+        fallback_log_filename
+    );
+}
+
+// Helper function to handle errors during stage2-init process
+// * `pre_privot_root` - bool to indicate whether handling err
+//
+// prior to pivot_root command being called. This is relevant
+// because if using fallback log mechanism, logs are still in memory
+// before the calling pivot_root. We attempt to setup log to tmpfs
+// in that case.
+fn stage2_init_err_handler(pre_pivot_root: bool, s2_config: &Stage2Config) -> ! {
```

Comment:
I would have used constants like PRE_PIVOT and POST_PIVOT to make the intent clearer. We change this later if there is consensus.

Reply:
i can address that

Reply:
I'm surprised that *all* of the callers of this function pass true for the `pre_pivot_root` parameter. Is this correct?